# Concatenate the reference genomes into a pan genome

The pooled populations of gastrointestinal nematodes are composed of ~99% of out species of interest Ostertagia ostertagi and Cooperia oncophora, they could not be split into seperate species populations pools before sequencing, therefore a pan-genome of both species needs to be created to align to, so that there is no minimise miss-alignment and alignment bias. 

## Concatenate reference genomes

Concatenate reference genomes script: concatenate_ref_genomes.sh 

    #!/bin/bash -l

    ############# SLURM SETTINGS #############
    #SBATCH --account=project0005
    #SBATCH --job-name=concatenate
    #SBATCH --output=%x-%J.out
    #SBATCH --error=%x-%J.err
    #SBATCH --time=0-00:30:00
    #SBATCH --mem=1G
    #SBATCH --ntasks=1
    #SBATCH --cpus-per-task=1
    #SBATCH --ntasks-per-node=1
    #SBATCH --mail-user=p.campbell.5@research.gla.ac.uk
    #SBATCH --mail-type=ALL

    ############# LOADING MODULES #############

    ############# MY CODE #############
    
    gunzip -c GCA_964213955.1_nxOstOste4.1_genomic.fna.gz  GCA_965643665.1_nxCooOnco1.hap1.1_genomic.fna.gz > combined_ref_genome.fa

## Sense check pan-genome so that all scaffolds have been combined 

Script for sequence and line counting: genome_stats.sh

    #!/bin/bash -l

     ############# SLURM SETTINGS #############
     #SBATCH --account=project0005
    #SBATCH --job-name=genome_stats
    #SBATCH --output=%x-%J.out
    #SBATCH --error=%x-%J.err
    #SBATCH --time=0-01:00:00
    #SBATCH --mem=1G
    #SBATCH --ntasks=1
    #SBATCH --cpus-per-task=1
    #SBATCH --ntasks-per-node=1
    #SBATCH --mail-user=p.campbell.5@research.gla.ac.uk
    #SBATCH --mail-type=ALL

    ############# LOADING MODULES #############

    ############# MY CODE #############

    # Output file
    OUTFILE="genome_statistics.txt"

    # Clear output file if it already exists
    : > "${OUTFILE}"

    # Function to print a header
    print_header () {
        echo "========================================" >> "${OUTFILE}"
        echo "$1" >> "${OUTFILE}"
        echo "========================================" >> "${OUTFILE}"
    }

    print_header "FASTA record counts (number of sequences)"

    echo "GCA_965643665.1_nxCooOnco1.hap1.1_genomic.fna.gz:" >> "${OUTFILE}"
    zgrep -c '^>' GCA_965643665.1_nxCooOnco1.hap1.1_genomic.fna.gz >> "${OUTFILE}"
    echo >> "${OUTFILE}"

    echo "GCA_964213955.1_nxOstOste4.1_genomic.fna.gz:" >> "${OUTFILE}"
    zgrep -c '^>' GCA_964213955.1_nxOstOste4.1_genomic.fna.gz >> "${OUTFILE}"
    echo >> "${OUTFILE}"

    echo "combined_ref_genome.fa:" >> "${OUTFILE}"
    grep -c '^>' combined_ref_genome.fa >> "${OUTFILE}"
    echo >> "${OUTFILE}"

    print_header "Line counts (total number of lines)"

    echo "GCA_965643665.1_nxCooOnco1.hap1.1_genomic.fna.gz:" >> "${OUTFILE}"
    zcat GCA_965643665.1_nxCooOnco1.hap1.1_genomic.fna.gz | wc -l >> "${OUTFILE}"
    echo >> "${OUTFILE}"

    echo "GCA_964213955.1_nxOstOste4.1_genomic.fna.gz:" >> "${OUTFILE}"
    zcat GCA_964213955.1_nxOstOste4.1_genomic.fna.gz | wc -l >> "${OUTFILE}"
    echo >> "${OUTFILE}"

    echo "combined_ref_genome.fa:" >> "${OUTFILE}"
    wc -l combined_ref_genome.fa >> "${OUTFILE}"

Output file: genome_statistics.txt

    ========================================
    FASTA record counts (number of sequences)
    ========================================
    GCA_965643665.1_nxCooOnco1.hap1.1_genomic.fna.gz:
    194

    GCA_964213955.1_nxOstOste4.1_genomic.fna.gz:
    118

    combined_ref_genome.fa:
    312

    ========================================
    Line counts (total number of lines)
    ========================================
    GCA_965643665.1_nxCooOnco1.hap1.1_genomic.fna.gz:
    5415361

    GCA_964213955.1_nxOstOste4.1_genomic.fna.gz:
    5090015

    combined_ref_genome.fa:
    10505376 combined_ref_genome.fa
    




